src/tests/ftest/pool/create_capacity.py

```diff
@@ -20,12 +22,284 @@ class PoolCreateCapacityTests(TestWithServers):
+        """Returns the available storage of a given NVMe namespace.
+
+        Args:
+            namespaces (dict): dmg json command output defining the NVMe namespaces.
+            namespace_id (int): Identifier of a namespace.
+            cluster_bytes (int): Size of an SPDK cluster.
+
+        Returns:
+            int: Available space of the given NVMe namespace.
+
+        """
+        for nsp in namespaces:
+            if nsp["id"] != namespace_id:
+                continue
+            return int(nsp["size"] / cluster_bytes) * cluster_bytes
+        self.fail(f"Namespace of id {namespace_id} could not be find")
+
+    def get_available_storage(self, storage_usage):
+        """Returns the largest available storage of the tiers storage.
+
+        Returns a two elements tuple defining the SCM and NVMe storage space which can  be used for
+        storing a pool.
+
+        Args
+            storage_usage (dict): dmg json command output defining the storage usage.
+
+        Returns:
+            tuple: SCM and NVMe usable storage.
+
+        """
+        scm_engine_bytes = {}
+        nvme_engine_bytes = {}
+        for host_storage in storage_usage["response"]["HostStorage"].values():
+
+            for scm_device in host_storage["storage"]["scm_namespaces"]:
+                rank = scm_device["mount"]["rank"]
+                if rank not in scm_engine_bytes:
+                    scm_engine_bytes[rank] = 0
+                scm_engine_bytes[rank] += scm_device["mount"]["avail_bytes"]
+
+            if host_storage["storage"]["nvme_devices"] is None:
+                continue
+
+            for nvme_device in host_storage["storage"]["nvme_devices"]:
+                if nvme_device["smd_devices"] is None:
+                    continue
+                namespace_id = 0
+                for smd_device in nvme_device["smd_devices"]:
+                    namespace_id += 1
+                    if smd_device["dev_state"] != "NORMAL":
+                        continue
+                    rank = smd_device["rank"]
+                    roles = smd_device["roles"]
+                    if not roles:
+                        # md-on-scm has only one implicit data role
+                        roles = "data"
+                    if rank not in nvme_engine_bytes:
+                        nvme_engine_bytes[rank] = {}
+                    if roles not in nvme_engine_bytes[rank]:
+                        nvme_engine_bytes[rank][roles] = 0
+                    namespaces = nvme_device["namespaces"]
+                    cluster_bytes = smd_device["cluster_size"]
+                    nvme_engine_bytes[rank][roles] += self.get_nsp_available_bytes(
+                        namespaces, namespace_id, cluster_bytes)
+
+        scm_bytes = sys.maxsize
+        for size in scm_engine_bytes.values():
+            scm_bytes = min(scm_bytes, size)
+        if scm_bytes == sys.maxsize:
+            scm_bytes = 0
```

Comment:
Is this trying to handle when `scm_engine_bytes.values()` is empty, or is it trying to handle when one of the values is greater than sys.maxsize?

Just FYI. This will give you the min from `scm_engine_bytes.values()`, or `0` if `values()` is empty
```suggestion
        scm_bytes = min(scm_engine_bytes.values(), default=0)
```

Reply:
You're welcome! A little more on min...
```
>>> # variable number of values
>>> min(3,1,2)
1
>>> # a list
>>> min([3,1,2])
1
>>> # a list and variable number of single values
>>> min(*[3,1,2], 0, 4)
0
```

Reply:
Fixed with commit 01eea32a89491faed5dbf7f03085398ff402c3c6